### PR TITLE
$PSCulture follows CurrentThread.CurrentCulture

### LINKED ISSUE #3831: Get-Culture and $PSCulture do not (consistently) reflect in-session switches to another culture
Note: Perhaps this is by design (although there's at least a buggy inconsistency), given that there's currently no _PowerShell_ mechanism for _in-session_ switching to another culture. If so, it should at least be documented.

Steps to reproduce
------------------

```powershell
"Initially:`t$((Get-Culture).Name)"
[cultureinfo]::currentculture = 'fr-FR'; "`$PSCulture`:`t$PSCulture"; "Get-Culture:`t$((Get-Culture).name)"; ".NET:`t`t$([cultureinfo]::currentculture.name)"
```

Note: 
*  If you try the above _interactively_, be sure to execute the 2nd line above _on a single line_.
* The output below:
  * assumes that `en-US` is the initial culture at session startup time.
  * differs between Windows and Unix: on Unix, `Get-Culture` reflects the switch, but `$PSCulture` doesn't, whereas on Windows neither does.

Expected behavior
-----------------

```none
en-US
fr-FR
fr-FR
fr-FR
```

Actual behavior
---------------

* macOS and Ubuntu

```none
Initially:	en-US
$PSCulture:	en-US
Get-Culture:	fr-FR
.NET:		fr-FR
```

* Windows

```none
Initially:	en-US
$PSCulture:	en-US
Get-Culture:	en-US
.NET:		fr-FR

```

Environment data
----------------

```powershell
PowerShell Core v6.0.0-beta (v6.0.0-beta.1) on macOS 10.12.5
PowerShell Core v6.0.0-beta (v6.0.0-beta.1) on Ubuntu 16.04.1 LTS
Windows PowerShell v5.1.14393.1198 on Microsoft Windows 10 Pro (64-bit; v10.0.14393)
```

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Now $PSCulture gets the value from CurrentThread.CurrentCulture

Close #3831

PowerShell Committee approved the change in #3831

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/4594
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: src/System.Management.Automation/engine/CultureVariable.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public override object Value
             get
```
Comment (PoshChan) [inline src/System.Management.Automation/engine/CultureVariable.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-macos`
$PSCulture.$PSCulture follows the current thread culture
```powershell
Expected strings to be the same, but they were different.
String lengths are both 5.
Strings differ at index 0.
Expected: 'ru-RU'
But was:  'en-US'
at <ScriptBlock>, /Users/vsts/agent/2.153.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Get-Culture.Tests.ps1: line 77
77:             $PSCulture | Should -BeExactly 'ru-RU'
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/CultureVariable.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
$PSCulture.$PSCulture follows the current thread culture
```powershell
Expected strings to be the same, but they were different.
String lengths are both 5.
Strings differ at index 0.
Expected: 'ru-RU'
But was:  'en-US'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Get-Culture.Tests.ps1: line 77
77:             $PSCulture | Should -BeExactly 'ru-RU'
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/CultureVariable.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
$PSCulture.$PSCulture follows the current thread culture
```powershell
Expected strings to be the same, but they were different.
String lengths are both 5.
Strings differ at index 0.
Expected: 'ru-RU'
But was:  'en-US'
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Get-Culture.Tests.ps1: line 77
77:             $PSCulture | Should -BeExactly 'ru-RU'
```

**Comment by iSazonov** [comment]
@PoshChan Please restart Macos

**Comment by PoshChan** [comment]
@iSazonov, I do not understand: Please restart Macos

Commands available in this repo for you:
  - `retry <target>` this will attempt to retry only the failed jobs for the target pipeline
  - `rebuild <target>` this will perform a complete rebuild of the target pipeline, `rerun` can be used in place of `rebuild`
    Supported values for \<target\> which can be a comma separated list are: `static`,`windows`,`macos`,`linux`,`all`
  - `get failures` this will attempt to get the latest failures for all of the target pipelines
  - `remind me in <value> <units>` this will create a reminder that will be posted after the specified duration
    \<value\> is a number, and \<units\> can be `minutes`, `hours`, or `days` (singular or plural)

**Comment by iSazonov** [comment]
@PoshChan Please retry Macos

**Comment by PoshChan** [comment]
@iSazonov, successfully started retry of `PowerShell-CI-macOS`

File: src/System.Management.Automation/engine/SessionState.cs [hunk 2]
```diff
@@ -348,11 +348,11 @@ internal void InitializeFixedVariables()
 
             // $PSCulture
             v = new PSCultureVariable();
-            this.GlobalScope.SetVariable(v.Name, v, asValue: false, force: true, this, CommandOrigin.Internal, fastPath: true);
+            this.GlobalScope.SetVariableForce(v, this);
```
Comment (TravisEz13) [inline src/System.Management.Automation/engine/SessionState.cs]: Why was this changed?

Comment (iSazonov) [reply src/System.Management.Automation/engine/SessionState.cs]: SetVariable() creates new variable by getting value as _constant_ from original variable but we need assign real object to always get fresh value.

**Comment by iSazonov** [comment]
@SteveL-MSFT @TravisEz13 Since the PR was approved by PowerShell Committee please review the code.

**Comment by iSazonov** [comment]
Rebased to get latest updates.

File: src/System.Management.Automation/engine/CultureVariable.cs (hunk 1 above)
Comment (PoshChan) [inline src/System.Management.Automation/engine/CultureVariable.cs]: @iSazonov, your last commit had 32 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

WSMan Config Provider.Set-Item tests.Set-Item on non-existent property should fail
```powershell
Expected an exception, with FullyQualifiedErrorId 'System.InvalidOperationException,Microsoft.PowerShell.Commands.SetItemCommand' to be thrown, but the FullyQualifiedErrorId was 'WsManError'. from D:\a\1\s\test\powershell\Modules\Microsoft.WSMan.Management\ConfigProvider.Tests.ps1:213 char:15
    +             { Set-Item $testPluginPath\foo "bar" } | Should -Throw -E ...
    +               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.WSMan.Management\ConfigProvider.Tests.ps1: line 213
213:             { Set-Item $testPluginPath\foo "bar" } | Should -Throw -ErrorId "System.InvalidOperationException,Microsoft.PowerShell.Commands.SetItemCommand"
```
WSMan Config Provider.Set-Item tests.Set-Item on plugin InitializationParameters 'PSVersion' property with '6.0.0' should succeed
```powershell
The WinRM client received an HTTP server error status (500), but the remote service did not include any other information about the cause of the failure. 
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.WSMan.Management\ConfigProvider.Tests.ps1: line 254
254:             Set-Item "$testPluginPath\InitializationParameters\$property" $value -WarningAction SilentlyContinue

```
WSMan Config Provider.Set-Item tests.Set-Item on plugin InitializationParameters 'psvERSION' property with '5.1' should succeed
```powershell
The WinRM client received an HTTP server error status (500), but the remote service did not include any other information about the cause of the failure. 
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.WSMan.Management\ConfigProvider.Tests.ps1: line 254
254:             Set-Item "$testPluginPath\InitializationParameters\$property" $value -WarningAction SilentlyContinue

```
WSMan Config Provider.Set-Item tests.Set-Item on plugin Quotas 'IdleTimeoutms' property with '61000' should succeed
```powershell
The WinRM client received an HTTP server error status (500), but the remote service did not include any other information about the cause of the failure. 
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.WSMan.Management\ConfigProvider.Tests.ps1: line 265
265:             Set-Item "$testPluginPath\Quotas\$property" $value -WarningAction SilentlyContinue

```
WSMan Config Provider.Set-Item tests.Set-Item on plugin Quotas 'iDLEtIMEOUTMS' property with '7200000' should succeed
```powershell
The WinRM client received an HTTP server error status (500), but the remote service did not include any other information about the cause of the failure. 
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.WSMan.Management\ConfigProvider.Tests.ps1: line 265
265:             Set-Item "$testPluginPath\Quotas\$property" $value -WarningAction SilentlyContinue

```

Comment (PoshChan) [inline src/System.Management.Automation/engine/CultureVariable.cs]: @iSazonov, your last commit had 36 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Error occurred in test script 'D:\a\1\s\test\powershell\Language\Operators\TernaryOperator.Tests.ps1'
```powershell
At D:\a\1\s\test\powershell\Language\Operators\TernaryOperator.Tests.ps1:15 char:37
+                 @{ Script = { $true ? 1 : 2 };  ExpectedValue = 1 }
+                                     ~
Unexpected token '?' in expression or statement.

At D:\a\1\s\test\powershell\Language\Operators\TernaryOperator.Tests.ps1:16 char:38
+                 @{ Script = { $true? ?1 :2 };   ExpectedValue = 2 }
+                                      ~~
Unexpected token '?1' in expression or statement.

At D:\a\1\s\test\powershell\Language\Operators\TernaryOperator.Tests.ps1:17 char:38
+                 @{ Script = { ${true}?1:2 };    ExpectedValue = 1 }
+                                      ~~~~
Unexpected token '?1:2' in expression or statement.

At D:\a\1\s\test\powershell\Language\Operators\TernaryOperator.Tests.ps1:18 char:33
+                 @{ Script = { 1 ? 1kb : 0xf };  ExpectedValue = 1kb }
+                                 ~
Unexpected token '?' in expression or statement.

At D:\a\1\s\test\powershell\Language\Operators\TernaryOperator.Tests.ps1:19 char:33
+                 @{ Script = { 0 ?1kb:0xf };     ExpectedValue = 15 }
+                                 ~~~~~~~~
Unexpected token '?1kb:0xf' in expression or statement.

At D:\a\1\s\test\powershell\Language\Operators\TernaryOperator.Tests.ps1:20 char:35
+                 @{ Script = { 's' ?1kb:0xf };   ExpectedValue = 1kb }
+                                   ~~~~~~~~
Unexpected token '?1kb:0xf' in expression or statement.

At D:\a\1\s\test\powershell\Language\Operators\TernaryOperator.Tests.ps1:21 char:37
+                 @{ Script = { $null ?1kb:0xf }; ExpectedValue = 15 }
+                                     ~~~~~~~~
Unexpected token '?1kb:0xf' in expression or statement.

At D:\a\1\s\test\powershell\Language\Operators\TernaryOperator.Tests.ps1:22 char:34
+                 @{ Script = { '' ?1kb:0xf };    ExpectedValue = 15 }
+                                  ~~~~~~~~
Unexpected token '?1kb:0xf' in expression or statement.

At D:\a\1\s\test\powershell\Language\Operators\TernaryOperator.Tests.ps1:25 char:39
+                 @{ Script = { 1,2,3,4 ? 'Core' : 'Desktop' };         ...
+                                       ~
Unexpected token '?' in expression or statement.

At D:\a\1\s\test\powershell\Language\Operators\TernaryOperator.Tests.ps1:26 char:42
+                 @{ Script = { @(1,2,3,4) ? 'Core' : 'Desktop' };      ...
+                                          ~
Unexpected token '?' in expression or statement.

Not all parse errors were reported.  Correct the reported errors and try again.
at <ScriptBlock>, D:\a\1\s\src\powershell-win-core\bin\Release\netcoreapp3.0\win7-x64\publish\Modules\Pester\4.8.0\Pester.psm1: line 1097
```
Set/New/Remove-Service cmdlet tests.New-Service can create a new service called 'testautomatic'
```powershell
Expected 'one', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Set-Service.Tests.ps1: line 217
217:             $service.displayname | Should -Be $(if($displayname){$displayname}else{$name})
```
Set/New/Remove-Service cmdlet tests.New-Service can create a new service called 'testmanual'
```powershell
Expected 'two', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Set-Service.Tests.ps1: line 217
217:             $service.displayname | Should -Be $(if($displayname){$displayname}else{$name})
```
Set/New/Remove-Service cmdlet tests.New-Service can create a new service called 'testdisabled'
```powershell
Expected 'testdisabled', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Set-Service.Tests.ps1: line 217
217:             $service.displayname | Should -Be $(if($displayname){$displayname}else{$name})
```
WSMan Config Provider.Set-Item tests.Set-Item on non-existent property should fail
```powershell
Expected an exception, with FullyQualifiedErrorId 'System.InvalidOperationException,Microsoft.PowerShell.Commands.SetItemCommand' to be thrown, but the FullyQualifiedErrorId was 'WsManError'. from D:\a\1\s\test\powershell\Modules\Microsoft.WSMan.Management\ConfigProvider.Tests.ps1:213 char:15
    +             { Set-Item $testPluginPath\foo "bar" } | Should -Throw -E ...
    +               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.WSMan.Management\ConfigProvider.Tests.ps1: line 213
213:             { Set-Item $testPluginPath\foo "bar" } | Should -Throw -ErrorId "System.InvalidOperationException,Microsoft.PowerShell.Commands.SetItemCommand"
```

**Comment by iSazonov** [comment]
@SteveL-MSFT @TravisEz13 Could you please continue the review?

**Review by TravisEz13 — APPROVED** [review]

**Comment by iSazonov** [comment]
@SteveL-MSFT Have you any concerns about the PR?

**Comment by iSazonov** [comment]
@SteveL-MSFT Friendly ping.

**Pull request merged**